### PR TITLE
Syncing style with pokecrystal

### PR DESCRIPTION
**Pull request by Rangi42**
Resolves issues #249 and #265.

### CODE REVIEW

**Comment by Rangi42** [comment]
@dannye Everything in #249 is taken care of.

**Comment by Rangi42** [comment]
@dannye Thank you! Please review the last commit and I'll merge this.

**Comment by Rangi42** [comment]
Note, home/pathfinding.asm is called that because its functions are all called from engine/overworld/pathfinding.asm.

File: home/joypad2.asm [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	cp LINK_STATE_BATTLING
+	jr z, .inLinkBattle
+	call WaitForTextScrollButtonPress
+	ld a, SFX_PRESS_AB
+	jp PlaySound
+.inLinkBattle
+	ld c, 65
+	jp DelayFrames
+
+INCLUDE "home/math.asm"
```
Comment (dannye) [inline home/joypad2.asm:97]: `home/joypad2.asm` seems to include a number of things not quite related to the joypad.

Comment (Rangi42) [reply home/joypad2.asm]: Now it's further split.

**Pull request merged**